### PR TITLE
V6 status indicator

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview
Created the new Status Indicator component

## Screenshots (if applicable)
image
image
image

## Testing

- [x] Updated Typescript types and/or component PropTypes
- [x] Added / modified component docs
- [x] Added / modified Storybook stories

### CODE REVIEW

**Comment by mshwery** [comment]
Did you mean to target `master`?

**Comment by colinlohner** [comment]
@mshwery nope! thanks

File: index.d.ts [hunk 1]
```diff
@@ -1626,6 +1626,15 @@ export declare const Stack: React.FC<StackProps>
 
 export declare const StackingContext: React.Context<number>
 
+export interface StatusIndicatorOwnProps extends TextOwnProps {
+  children?: React.ReactNode
```
Comment (mshwery) [inline index.d.ts:1630]: iirc you shouldn't ever have to specify `children`

File: yarn.lock [hunk 2]
```diff
@@ -1791,6 +1791,25 @@
   dependencies:
     babel-runtime "^6.26.0"
 
+"@storybook/addon-storysource@^5.3.19":
```
Comment (mshwery) [inline yarn.lock:1794]: Wonder if we can add something that reveals the props and let's you modify them too? I recall something like this for storybook.

File: src/status-indicator/src/StatusIndicator.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const StatusIndicator = memo(
+  forwardRef(function StatusIndicator(props, ref) {
+    const { children, disabled, color = 'info', ...rest } = props
+    const dotColor = useIconColor(color === 'none' ? 'info' : color)
+
+    return (
+      <Text display="inline-flex" alignItems="center" ref={ref} {...rest}>
+        <Box
+          flexShrink={0}
+          marginRight={8}
```
Comment (mshwery) [inline src/status-indicator/src/StatusIndicator.js:16]: We should source this from the theme spacing (or at least from the scale)?

File: src/status-indicator/src/StatusIndicator.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          borderRadius="50%"
+          height={7}
+          width={7}
```
Comment (mshwery) [inline src/status-indicator/src/StatusIndicator.js:19]: Should this be controllable via `dotSize` or similar?

File: yarn.lock (hunk 2 above)
Comment (colinlohner) [reply yarn.lock:1794]: Yeah, gonna add knobs in somepoint soon. but its a little more involved

**Pull request merged**